writing-guide.md

```diff
@@ -0,0 +1,86 @@
+  - **Essentials**: It should take no longer than 5 hours to read the Essentials, though shorter is better. Its goal is to provide the 20% of knowledge that will help users handle 80% of use cases. It can link to more advanced guides and the API, though these should be avoided in most cases and when provided, context should be provided so that users know they should  follow that link on their first reading. Otherwise, many users end up exhausting their cognitive capacity link-hopping, trying to fully learn every aspect of a feature before moving on, and as a result, never finish that first read-through of the Essentials. Remember that a smooth read is more important than being thorough. We want to give people the information they need to avoid a frustrating experience, but they can always come back and read further, or Google a less common problem when they encounter it.
+  - **Advanced**: While the Essentials helps people handle ~80% of use cases, subsequent guides help get users to 95% of use cases, plus more detailed information on non-essential features (e.g. transitions, animations), more complex convenience features (e.g. mixins, custom directives), and dev experience improvements (e.g. JSX, plugins). The final 5% of use cases that are more niche, complex, and/or prone to abuse will be left to the cookbook and API reference, which can be linked to from these advanced guides.
+- **Reference/API**: Provide a complete list of features, including type information, descriptions of the problem each solves, examples of every combination of options, and links to guides, cookbook recipes, and other internal resources providing more detail. Unlike other pages, this one is not meant to be read top-to-bottom, so plenty of detail can be provided. These references must also be more easily skimmable than the guides, so the format should be closer to dictionary entries than the story-telling format of the guides.
+- **Migrations**:
+  - **Versions**: When important changes are made, it's useful to include a full list of changes, including a detailed explanation of why the change was made and how to migrate their projects.
+  - **From other projects**: How does this software compare to similar software? This is important to help users understand what additional problems we might solve or create for them, and to what extent they can transfer knowledge they already have.
+- **Style Guide**: There are necessarily some key pieces in development that need a decision, but are not core to the API. The style guide provides educated, opinionated recommendations to help guide these decisions. They shouldn't be followed blindly, but can help teams save time by being aligned on smaller details.
+- **Cookbook**: Recipes in the cookbook are written with some assumption of familiarity with Vue and its ecosystem. Each is a highly structured document that walks through some common implementation details that a Vue dev might encounter.
+
+## Writing & Grammar
```

Comment:
_Very_ nitpicky: Let's avoid using abbreviations/symbols like "&" as well and replace it with "and."

Reply:
Hm, what's your reasoning? And are you thinking everywhere, or only in non-titles?

Reply:
My reasoning is actually pretty simple: We're discouraging abbreviations elsewhere (in this same document), right? "&" to me is another abbreviation 😛

Reply:
My reasoning for avoiding abbreviations is that some people might not be familiar with them, or even when familiar, might have to translate in their heads to something _more_ familiar. I don't have the same worry with symbols that are on everyone's keyboards, like `&`, `%`, `#`, and `@`, even though you could argue that all of them started as abbreviations. By now, they're essentially punctuation. In my experience, even those who have never used a computer are familiar with them. Is your experience different?

Reply:
I would argue that it's easier to draw the line between _full words and abbreviations_ than between _less and more familiar_, but it's not extremely important. I guess I can live with either.

Reply:
I would argue that it's easier to draw the line between full words and abbreviations than between less and more familiar

I agree, but I also think most people are not thinking of symbols as abbreviations. Plus, if it's so universal that it's on every standard keyboard (that I've seen, at least) and even crosses language barriers, then I think it's OK. 😄 I just added a note to address this.